### PR TITLE
Add-bitbucketRepositoryRestriction now properly removes branch_match_kind property

### PR DESCRIPTION
**Pull request by variableresistor**
Select-Object returns all properties in PowerShell version 5.1 when the caller specifies the -ExcludeProperty parameter, but does not specify the -Properties parameter.

### CODE REVIEW

**Comment by variableresistor** [comment]
I actually don't know how the tests are working. Perhaps someone from the team can help me figure it out. :-) I've spent at least an hour trying to figure it out on my own.

File: Tests/BranchRestriction.Tests/Add-BitbucketRepositoryBranchRestriction.Tests.ps1 [hunk 1]
```diff
@@ -32,6 +32,12 @@ Describe 'Add-BitbucketRepositoryBranchRestriction' {
                 ($MergeCheck | Select-Object -ExcludeProperty branch_type | ConvertTo-Json -Depth 2 -Compress) -eq $Body
             }
         }
+
+        It "Removes the branch_type property when branch_match_kind equals 'glob'" {
```
Comment (variableresistor) [inline Tests/BranchRestriction.Tests/Add-BitbucketRepositoryBranchRestriction.Tests.ps1:36]: Here I'm making sure that the property doesn't exist after I convert the body from a JSON object into a PSObject

**Comment by eelkram** [comment]
I've updated the module with a bunch of changes for the team -> workspace migration.  Can you update your PR and merge in the changes and then I'll take a look.

**Comment by variableresistor** [comment]
I'm just going to delete my fork and re-submit a PR; I'm used to this on Azure DevOps and don't know my way around Github PRs yet.

**Pull request closed without merging**